### PR TITLE
Implement PyUpgrade in pre-commit

### PR DESCRIPTION
**Pull request by kedhammar**
- [x] Add PyUpgrade to pre-commit config and dev requirements.
- [x] `CHANGELOG.md` is updated

### CODE REVIEW

**Comment by kedhammar** [comment]
Mentioned in #2200

**Review by mashehu — APPROVED** [review]

File: requirements-dev.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ Sphinx
 sphinx-rtd-theme
 mypy
 types-PyYAML
+pyupgrades
```
Comment (fabianegli) [inline requirements-dev.txt:11]: Typo...
```suggestion
pyupgrade
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -11,3 +11,7 @@ repos:
     rev: "v2.7.1"
     hooks:
       - id: prettier
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v3.15.0
+    hooks:
+    -   id: pyupgrade
```
Comment (fabianegli) [inline .pre-commit-config.yaml:17]: We should add the Python version we want to upgrade to as explained here: https://pre-commit.com/#passing-arguments-to-hooks and https://pre-commit.com/#pre-commit-autoupdate

```suggestion
    -   id: pyupgrade
        args: [--py38-plus]
```

**Review by fabianegli — CHANGES_REQUESTED** [review]
For Pyupgrade not to introduce changes incompatible with our supported Python versions, it requires the flag indicating the oldest supported Python version.

**Comment by mirpedrol** [comment]
@nf-core-bot fix linting

**Comment by mirpedrol** [comment]
@fabianegli are you happy to merge this once tests pass?

**Review by fabianegli — APPROVED** [review]

**Pull request merged**